allennlp/data/dataset_readers/reading_comprehension/triviaqa.py

```diff
@@ -0,0 +1,162 @@
+        logger.info("Opening base tarball file at %s", self._base_tarball_path)
+        base_tarball = tarfile.open(cached_path(self._base_tarball_path), 'r')
+        if 'unfiltered' in file_path:
+            logger.info("Opening unfiltered tarball file at %s", self._unfiltered_tarball_path)
+            unfiltered_tarball = tarfile.open(cached_path(self._unfiltered_tarball_path), 'r')
+            logger.info("Loading question file from tarball")
+            data_json = json.loads(unfiltered_tarball.extractfile(file_path).read().decode('utf-8'))
+        else:
+            logger.info("Loading question file from tarball")
+            path = os.path.join('qa', file_path)
+            data_json = json.loads(base_tarball.extractfile(path).read().decode('utf-8'))
+
+        logger.info("Reading the dataset")
+        instances = []
+        for question_json in tqdm(data_json['Data']):
+            question_text = question_json['Question']
+            question_tokens = self._tokenizer.tokenize(question_text)
+
+            evidence_files: List[List[str]] = []  # contains lines from each evidence file
+            if 'web' in file_path:
+                for result in question_json['SearchResults']:
+                    filename = result['Filename']
+                    evidence_file = base_tarball.extractfile(os.path.join("evidence", "web", filename))
+                    evidence_files.append([line.decode('utf-8') for line in evidence_file.readlines()])
+            else:
+                for result in question_json['EntityPages']:
+                    filename = result['Filename']
+                    evidence_file = base_tarball.extractfile(os.path.join("evidence", "wikipedia", filename))
+                    evidence_files.append([line.decode('utf-8') for line in evidence_file.readlines()])
+
+            answer_json = question_json['Answer']
+            answer_texts = answer_json['NormalizedAliases'] + answer_json.get('HumanAnswers', [])
+            for paragraph in self.pick_paragraphs(evidence_files, question_text, answer_texts):
+                instance = self.text_to_instance(question_text, paragraph, answer_texts, question_tokens)
+                if instance:
+                    instances.append(instance)
+        if not instances:
+            raise ConfigurationError("No instances were read from the given filepath {}. "
+                                     "Is the path correct?".format(file_path))
+        return Dataset(instances)
+
+    def pick_paragraphs(self,
+                        evidence_files: List[List[str]],
+                        question: str = None,
+                        answer_texts: List[str] = None) -> List[str]:
+        """
+        Given a list of evidence documents, return a list of paragraphs to use as training
+        examples.  Each paragraph returned will be made into one training example.
+
+        To aid in picking the best paragraph, you can also optionally pass the question text or the
```

Comment:
This seems out of date - the arguments are unused. Seems like you can drop the pylint annotations make the method static and remove these arguments.

Reply:
It's actually forward-looking.  I'm going to add in some options in the next PR that actually use these arguments.

Reply:
🔮